include/seastar/core/gate.hh

```diff
@@ -246,29 +262,35 @@ public:
                 _g = nullptr;
             }
         }
-
-    private:
-#ifdef SEASTAR_GATE_HOLDER_DEBUG
-        void debug_hold_gate() noexcept {
-            ++_g->_holders;
-        }
-
-        void debug_release_gate() noexcept {
-            --_g->_holders;
-        }
-#else   // SEASTAR_GATE_HOLDER_DEBUG
-        void debug_hold_gate() noexcept {}
-        void debug_release_gate() noexcept {}
-#endif  // SEASTAR_GATE_HOLDER_DEBUG
     };
 
     /// Get a RAII-based gate::holder object that \ref enter "enter()s"
     /// the gate when constructed and \ref leave "leave()s" it when destroyed.
     holder hold() {
         return holder(*this);
     }
+
+private:
+#ifdef SEASTAR_GATE_HOLDER_DEBUG
+    using holders_list_t = boost::intrusive::list<holder,
+        boost::intrusive::member_hook<holder, holder::member_hook_t,
+        &holder::_link>,
```

Comment:
nit: this `&holder::_link>,` should be on previous line

Reply:
why? having each param on it's own line is quite clear.
maybe the sin here is using single indent rather than double :)

Reply:
Well, yes, so it should be either

```
    using holders_list_t = boost::intrusive::list<holder,
        boost::intrusive::member_hook<holder, holder::member_hook_t, &holder::_link>,
        boost::intrusive::constant_time_size<false>>;
```

or

```
    using holders_list_t = boost::intrusive::list<holder,
        boost::intrusive::member_hook<holder,
            holder::member_hook_t,
            &holder::_link>,
        boost::intrusive::constant_time_size<false>>;

```

Reply:
ah, got your point.
clang-format to the rescue! :)

Reply:
fixed in v2